includes/helper-functions.php

```diff
@@ -857,3 +857,83 @@ function edac_database_table_count( $table ) {
 
 	return $count;
 }
+
+/**
+ * Add a scheme to file it looks like a url but doesn't have one.
+ *
+ * @since 1.10.1
+ *
+ * @param  string $file The filename.
+ * @param string $site_protocol The site protocol. Default is 'https'.
+ *
+ * @return string The filename unchanged if it doesn't look like a URL, or with a scheme added if it does.
+ */
+function edac_url_add_scheme_if_not_existing( string $file, string $site_protocol = 'https' ): string {
+	if ( str_starts_with( $file, $site_protocol ) || ! str_starts_with( $file, '//' ) ) {
+		return $file;
+	}
+
+	return "{$site_protocol}:{$file}";
+}
+
+/**
+ * Requests the headers of a URL to check if it exists.
+ *
+ * @since 1.10.1
+ *
+ * @param string $url the url to check.
+ * @return bool
+ */
+function edac_url_exists( string $url ): bool {
+
+	$response = wp_remote_head( $url );
+
+	if (
+		is_wp_error( $response ) ||
+		( // Check if the response code is not in the 2xx range.
+			wp_remote_retrieve_response_code( $response ) < 200 ||
+			wp_remote_retrieve_response_code( $response ) > 299
+		)
+	) {
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Get a file from local or remote source as a binary file handle.
+ *
+ * @since 1.10.1
+ *
+ * @param string $filename The file location, either local or a remote URL.
+ * @return resource|bool The file binary string or false if the file could not be opened.
+ */
+function edac_get_file_opened_as_binary( string $filename ) {
+	// If the file name looks like a url without a scheme add https and check if the file exists.
+	if ( ! str_starts_with( $filename, '//' ) ) {
+		$file = $filename;
+	} else {
+		$file       = edac_url_add_scheme_if_not_existing( $filename );
+		$url_exists = edac_url_exists( $file );
+		// if the https version does not exist, try http.
+		if ( false === $url_exists ) {
+			$file       = edac_url_add_scheme_if_not_existing( $filename, 'http' );
+			$url_exists = edac_url_exists( $file );
+		}
```

Comment:
I thought about this over the weekend and I want to just do away with the nested if logic here and instead assume it will always have only what the site protocal is since the page it loads on would dictate it. Thoughts on dropping the internal if here that falls back to `http`?

Reply:
Yes, we can remove this logic check.